File: core/src/com/quillraven/game/ecs/system/AnimationSystem.java
package com.quillraven.game.ecs.system;

import com.badlogic.ashley.core.ComponentMapper;
import com.badlogic.ashley.core.Entity;
import com.badlogic.ashley.core.Family;
import com.badlogic.ashley.systems.IteratingSystem;
import com.quillraven.game.ecs.component.AnimationComponent;

public class AnimationSystem extends IteratingSystem {
    private final ComponentMapper<AnimationComponent> aniCmpMapper;

    public AnimationSystem(final ComponentMapper<AnimationComponent> aniCmpMapper) {
        super(Family.all(AnimationComponent.class).get());

        this.aniCmpMapper = aniCmpMapper;
    }

    @Override
    protected void processEntity(final Entity entity, final float deltaTime) {
        aniCmpMapper.get(entity).aniTimer += deltaTime;
    }
}

File: core/src/main/kotlin/com/github/quillraven/mysticgarden/system/AnimationSystem.kt
package com.github.quillraven.mysticgarden.system

import com.github.quillraven.fleks.Entity
import com.github.quillraven.fleks.IteratingSystem
import com.github.quillraven.fleks.World.Companion.family
import com.github.quillraven.mysticgarden.component.Animation
import com.github.quillraven.mysticgarden.component.Disable
import com.github.quillraven.mysticgarden.component.Render

class AnimationSystem : IteratingSystem(family { all(Animation, Render).none(Disable) }) {

    override fun onTickEntity(entity: Entity) {
        val aniCmp = entity[Animation]
        val (animation, speed, stateTime, mode) = aniCmp

        animation.playMode = mode
        val frame = animation.getKeyFrame(stateTime)
        aniCmp.stateTime += deltaTime * speed

        entity[Render].sprite.setRegion(frame)
    }
}